uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,447 @@ define([
     templateUrl: ['settings', function (settings) {
       return settings.pathTpl + 'components/manager-leave-requests.html';
     }],
-    controllerAs: 'requests',
-    controller: ['$log', controller]
+    controllerAs: 'ctrl',
+    controller: ['$log', '$q', 'Contact', 'AbsencePeriod', 'AbsenceType', 'LeaveRequest',
+      'OptionGroup', controller]
   });
 
 
-  function controller($log) {
+  function controller($log, $q, Contact, AbsencePeriod, AbsenceType, LeaveRequest, OptionGroup) {
     $log.debug('Component: manager-leave-requests');
+
+    var vm = Object.create(this),
+      allStatus = {
```

Comment:
I don't like this to be honest. You are supposed to have a `statuses` array that contain every status for the filters (see my comment below, where i point out that you have `statuses` as a global var), so why don' you initialize it directly on `statuses` instead of having an ad-hoc variable? You could use the `name` of each status as the keys in `statuses` so it's easy to retrieve them
```js
var statuses = { 'all': { name: 'all', label: 'All' } };

// Later..
leaveRequestFilters: {
  selectedUsers: null,
  selectedPeriod: null,
  selectedAbsenceTypes: null,
  leaveStatus: statuses.all,
  pending_requests: false
}